app/tests/unit/confirm-dialog.spec.ts

```diff
@@ -0,0 +1,40 @@
+import { shallowMount, mount, Wrapper } from '@vue/test-utils'
+import ConfirmationDialog from '@/components/common/ConfirmationDialog.vue';
+import Button from '@/components/common/Button.vue';
+
+describe('ConfirmationDialog.vue', () => {
+  let wrapper:Wrapper<Vue>;
+
+  beforeEach(() => {
+    wrapper = mount(ConfirmationDialog);
+  });
+
+  afterEach(() => {
+    wrapper.destroy();
+  });
+
+  it('emits confirm event when Confirm button is clicked', async () => {
+    await wrapper.findComponent(Button).trigger('click');
+    await wrapper.vm.$nextTick();
```

Comment:
You would either need to `await` for the async `trigger` method or `wrapper.vm.$nextTick()`, no need to have both.

Reply:
I've removed nextTicks where it was possible. However, in some cases, tests don't pass if I remove one of await or await nextTick.

Reply:
For example, in file user-info-form.spec.ts in test "adds user when form is submitted for "add" action": ```
await wrapper.find('form').trigger('submit.prevent');
    await wrapper.vm.$nextTick();
```
When i delete one of the awaits, real data differs from expected.

Reply:
In this particular case `await wrapper.vm.$nextTick()` makes sense indeed, due to the asynchronous behavior of mockResolvedValueOnce, where we have a Promise as a result. It ensures any promises are settled before the assertion is made.

Reply:
Thanks for the info!